### PR TITLE
Minor memory leak fixes

### LINKED ISSUE #6513: Fix memleak introduced in PDB
introduced in 75dd4536fa3980041b22cf9da2863f0c0bbe2142

this requires some refactoring and code cleanup of the PDB parser.

### PR DESCRIPTION
**Pull request by nartes**
#6513

### CODE REVIEW

File: libr/bin/pdb/omap.c [hunk 1]
```diff
@@ -112,6 +112,14 @@ int omap_remap(void *stream, int address) {
...
 	pos = binary_search (omap_stream->froms, address, 0, (len));
 
+	// TODO(nartes): it might introduce seg fault, cause we might peek into
+	// the bytes before allocated area, i.e. omap_stream->froms[-1]
+	//
+	// if (pos == -1)
+	// {
+	//		return -1;
+	// }
+	assert (pos >= 0);
```
Comment (Maijin) [inline libr/bin/pdb/omap.c]: An assert, you mad?

Comment (Maijin) [reply libr/bin/pdb/omap.c]: https://travis-ci.org/radare/radare2/jobs/288191151#L864

File: libr/bin/bin.c [hunk 2]
```diff
@@ -930,7 +930,22 @@ R_API int r_bin_reload(RBin *bin, int fd, ut64 baseaddr) {
 	if (!bf) {
 		return false;
 	}
+
+	// TODO(nartes):  the_obj_list might be leaked along the way.
```
Comment (Maijin) [inline libr/bin/bin.c]: If everyone add 100 lines of TODO in the core it's quickly going to be a mess

File: libr/bin/pdb/pdb.c [hunk 3]
```diff
@@ -802,7 +802,7 @@ void build_command_field(ELeafType lt, char **command_field) {
 		if (!(*command_field)) {
 			break;
 		}
-		strcpy (*command_field, "pf.");
+		strcpy (*command_field, "pf");
```
Comment (Maijin) [inline libr/bin/pdb/pdb.c]: Are you sure about that ? Can you provide a testcase?

File: libr/bin/pdb/omap.c (hunk 1 above)
Comment (nartes) [reply libr/bin/pdb/omap.c]: Well, partial compilation didn't desire to run gcc extra time. I thought you had that sort of a macro. Well I'll fix it)

File: libr/bin/pdb/pdb.c (hunk 3 above)
Comment (nartes) [reply libr/bin/pdb/pdb.c]: https://gist.github.com/anonymous/57e419726335f709d6d00cd733364408
Here is a scripted gdb session that shows a memory corruption if the patch will not be applied.

Comment (Maijin) [reply libr/bin/pdb/pdb.c]: No but the pf and pf. is supposed to have a purpose it's not about a memory corruption, if you break the tool for fixing memory corruption it's pointless :/

Comment (nartes) [reply libr/bin/pdb/pdb.c]: Well then we should allocate the memory for pf. not pf. The problem is that memory doesn't much the amount of bytes copied. And alocate smaller amount than we use.)

**Pull request merged**